### PR TITLE
SSCS-10227 case flag mapping

### PR DESCRIPTION
**Pull request by zikrurR**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SSCS-10227

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/model/DescendantCaseData.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package uk.gov.hmcts.reform.sscs.model;
+
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+
+import java.util.Map;
+
+@EqualsAndHashCode(callSuper = true)
+@Builder (builderMethodName = "DescendantCaseDataBuilder")
+public class DescendantCaseData extends uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData {
```
Comment (dattipoeHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/model/DescendantCaseData.java:10]: This still needs changing

File: src/main/java/uk/gov/hmcts/reform/sscs/service/CaseFlagMappingService.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.helper.FlagCode;
+import uk.gov.hmcts.reform.sscs.model.DescendantCaseData;
+import uk.gov.hmcts.reform.sscs.model.HearingWrapper;
+
+import java.util.Map;
+import javax.validation.constraints.NotNull;
+
+@Service
```
Comment (dattipoeHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/CaseFlagMappingService.java:13]: I don't think this should be a service, it should be a static utility class with no state

File: src/main/java/uk/gov/hmcts/reform/sscs/service/CaseFlagMappingService.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+    public DescendantCaseData getCaseFlagMapping(@NotNull HearingWrapper hearingWrapper) {
+        SscsCaseData caseData = hearingWrapper.getOriginalCaseData();
+        return DescendantCaseData
+            .DescendantCaseDataBuilder()
+            .caseFlags(mapCaseFlags(caseData))
+            .build();
+    }
+
+    public Map<String, String> mapCaseFlags(SscsCaseData caseData) {
```
Comment (dattipoeHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/CaseFlagMappingService.java:28]: There's still no need for a map. The enum has all the information you need by itself

File: src/main/java/uk/gov/hmcts/reform/sscs/service/CaseFlagMappingService.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        mapSignLanguageType(caseData);
+        mapArrangements(caseData);
+        confidentialCase(caseData);
+        dwp(caseData);
+        urgentCase(caseData);
+        adjournCaseInterpreterLanguage(caseData);
+        return this.caseFlags;
+    }
+
+    private void mapSignLanguageType(SscsCaseData caseData) {
```
Comment (dattipoeHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/CaseFlagMappingService.java:38]: You didn't address the null or boolean issues from the previous PR

**Review by dattipoeHMCTS — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/FlagCode.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    DISABLED_ACCESS("RA0019"),
+    SIGN_LANGUAGE_TYPE("RA0042"),
+    HEARING_LOOP("RA0043"),
+    IS_CONFIDENTIAL_CASE("PF0004"),
+    DWP_UCB("PF0007"),
+    DWP_PHME("CF0003"),
+    URGENT_CASE("CF0007"),
+    AD_JOURN_CASE_INTERPRETER_LANGUAGE("PF0015");
+
+    private final String flagCode;
```
Comment (wmarshid) [inline src/main/java/uk/gov/hmcts/reform/sscs/helper/FlagCode.java:17]: Could we rename this to `code` instead?

File: src/main/java/uk/gov/hmcts/reform/sscs/model/DescendantCaseData.java (hunk 1 above)
Comment (wmarshid) [reply src/main/java/uk/gov/hmcts/reform/sscs/model/DescendantCaseData.java:10]: Echoing Del - not sure why this class is needed

File: src/main/java/uk/gov/hmcts/reform/sscs/service/CaseFlagMappingService.java (hunk 3 above)
Comment (wmarshid) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/CaseFlagMappingService.java:28]: Echoing Del - we could just add the relevant enums to a list on the object and the calling code can pull the code from the enum directly.

File: src/test/java/uk/gov/hmcts/reform/sscs/service/CaseFlagMappingServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+import uk.gov.hmcts.reform.sscs.ccd.domain.YesNo;
+import uk.gov.hmcts.reform.sscs.model.DescendantCaseData;
+import uk.gov.hmcts.reform.sscs.model.HearingWrapper;
+
+import java.util.List;
+import java.util.Map;
+
+@SpringBootTest
+@RunWith(SpringRunner.class)
+public class CaseFlagMappingServiceTest {
```
Comment (wmarshid) [inline src/test/java/uk/gov/hmcts/reform/sscs/service/CaseFlagMappingServiceTest.java:22]: Using an integration test for this is a bit overkill.. unit test should suffice.

**Review by wmarshid — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/FlagCode.java (hunk 5 above)
Comment (zikrurR) [reply src/main/java/uk/gov/hmcts/reform/sscs/helper/FlagCode.java:17]: oops

**Pull request closed without merging**